packages/interactivity/src/directives.tsx

```diff
@@ -286,9 +158,10 @@ export default () => {
 							`The value of data-wp-context in "${ namespace }" store must be a valid stringified JSON object.`
 						);
 					}
-					updateContext(
+					deepMerge(
 						currentValue.current,
-						deepClone( value ) as object
+						deepClone( value ) as object,
+						false
```

Comment:
The idea here is to use `deepMerge()` with the `override` option set to `false`. This algorithm follows the same logic as `updateContext` but does not manipulate the reactive object directly and only updates the signals created beforehand.

That way, we avoid `wp-context` subscribing unnecessarily to the modified context and prevent wasteful signal instantiation. 🙂

Reply:
ok, I see. Upon further inspection the logic does seem **almost** the same. The only 2 differences I see is:

1. We do not use `peek()` anymore. This I assume is because we the context is not wrapped in proxies anymore like you mentioned [here](https://github.com/WordPress/gutenberg/pull/64713#discussion_r1750588651).

2. I see that we're now assigning empty object here:
https://github.com/WordPress/gutenberg/blob/ce56700594dd7cb8b1794488e2c010706d310b81/packages/interactivity/src/proxies/state.ts#L298-L301

   but [previously](https://github.com/WordPress/gutenberg/pull/64713/files#diff-118c339960e17dbd5b9434e6a2eec6851c8b8f68b6ace6faf4e57603850eb169L147-L149) we used to re-assign the property:

   ```js
   } else if ( ! ( k in target ) ) {
     target[ k ] = source[ k ];
   }
   ```

Reply:
> 1. We do not use `peek()` anymore. This I assume is because we the context is not wrapped in proxies anymore like you mentioned [here](https://github.com/WordPress/gutenberg/pull/64713#discussion_r1750588651).

Exactly. 🙂 



> 2. I see that we're now assigning empty object here [...] but [previously](https://github.com/WordPress/gutenberg/pull/64713/files#diff-118c339960e17dbd5b9434e6a2eec6851c8b8f68b6ace6faf4e57603850eb169L147-L149) we used to re-assign the property:

That was wrong. We don't want to copy a plain object's reference from `source` when calling `deepMerge`.